### PR TITLE
Remove time-only latent rules

### PR DESCRIPTION
**Pull request by gabrielelanaro**
In this PR we remove the latent rules related to time, so that "times only" are represented without anchoring to a particular date which isn't implied by the string (in contrast with `Today`, where the expression refers to the reference date).

@sebastianmika  I believe we need to move away from travis

### CODE REVIEW

File: ctparse/nb_scorer.py [hunk 1]
```diff
@@ -64,7 +64,10 @@ def score_final(
         # NOTE: the prediction is log-odds, or logit
         model_score = pred[0][1] - pred[0][0]
 
-        return model_score + len_score
+        # We want the len_score to always take precedence. I believe a logit won't go up more
```
Comment (sebastianmika) [inline ctparse/nb_scorer.py:67]: As long as we can come up with a way to translate this into a scalar for the `CTParse` the internal changes would be rather small.
How come we "know" that we should first order by length and then the model?

File: ctparse/time/auto_corpus.py [hunk 2]
```diff
@@ -742,7 +742,7 @@
         ["12.06. um 13.15-14:15"],
     ],
     [
-        "Time[]{2016-11-03 08:15 (X/X)}",
+        "Interval[]{2016-11-03 08:15 (X/X) - None}",
```
Comment (sebastianmika) [inline ctparse/time/auto_corpus.py:745]: Interesting - a bug in the training data, how unfortunate

File: ctparse/time/corpus.py [hunk 3]
```diff
@@ -337,7 +337,7 @@
     ),
     # ruleTODTOD, ruleLatentTimeInterval
```
Comment (sebastianmika) [inline ctparse/time/corpus.py:338]: ~~`ruleLatentTimeInterval`~~

**Review by sebastianmika — COMMENTED** [review]
If "no latent dates" becomes the new behaviour we should also upgrade the docs/README.

This should become a major version change - unless we find a way to support both behaviours. I could still imagine situations where forced grounding on `today` would be useful. But it might be problematic to support that in the same model. So probably not worth the effort.

File: ctparse/nb_scorer.py (hunk 1 above)
Comment (gabrielelanaro) [reply ctparse/nb_scorer.py:67]: Simple Answer: One of the reason is that with this length scaling we get rid of an important regression while still maintaining performance. This comes up in simple expressions

Complicated answer: I believe to be the reason for this effect

```
15.10.2020 (ref date 14 oct 2020)
15.10 (ref date 14 oct 2020)
```

Expression 1 will match both the rule `15 October + latent_year` and `15 October + 2020`, while expression 2 will match only `15 October + latent_year`

Since in our dataset we mostly have that the reference date is same year as the one in the time expression, what happens is that the probability of the sequence that end in latent year gets a larger probability score than the one with the year, and the `len_score` is too small to deal with it (a half match would have a len score log(0.5) versus a full match log(1) which is only 0.3, logits for those highly imbalanced sequences can go 15-16).

Why would it appear in this particular PR? I believe it is because this affects the number of "negatives" since we remove certain productions for the strings of type `15.10`

**Comment by gabrielelanaro** [comment]
@sebastianmika PTAL
